packages/material-ui/src/LinearProgress/LinearProgress.d.ts

```diff
@@ -7,6 +7,7 @@ export interface LinearProgressProps
   value?: number;
   valueBuffer?: number;
   variant?: 'determinate' | 'indeterminate' | 'buffer' | 'query';
+  styles?: React.CSSProperties;
```

Comment:
This is wrong. `styles`, in our case, is not just a simple `React.CSSProperties`. For example, in this case, we have `bar1` and `bar2` properties in `styles` object, and for `bar1` or `bar2`, they are of type `React.CSSProperties`. So, the type definitions need to reflect this, and what I suggested you to do fulfills this requirement.

Reply:
@adeelibr 
I would do the following:
```ts
export type LinearProgressStylesKeys =
  | 'bar1'
  | 'bar2';
```
and then, `styles: React.CSSProperties;` can be changed to
```ts
styles?: { [P in LinearProgressStylesKeys]?: React.CSSProperties };
```